cmd/kuberhealthy/kuberhealthy.go

```diff
@@ -334,24 +334,22 @@ func (k *Kuberhealthy) addExternalChecks() error {
 		log.Infoln("Enabling external check:", r.Name)
 		c := external.New(kubernetesClient, &r, khCheckClient, khStateClient, externalCheckReportingURL)
 
-		// parse the run interval string from the custom resource and setup the run interval
+		// parse the run interval string from the custom resource and setup the run interval, error if interval not set properly
 		c.RunInterval, err = time.ParseDuration(r.Spec.RunInterval)
 		if err != nil {
-			log.Errorln("Error parsing duration for check", c.CheckName, "in namespace", c.Namespace, err)
-			log.Errorln("Defaulting check to a runtime of ten minutes.")
-			c.RunInterval = DefaultRunInterval
+			errorMessage := "Error parsing duration for check " + c.CheckName + " in namespace " + c.Namespace + err.Error()
+			log.Errorln(errorMessage)
+			return errors.New(errorMessage)
```

Comment:
If we return here, then we halt adding external checks.  I would prefer that when we have an external check configuration that isn't good (for any reason, including the RunInterval), we just log an error state for that specific check.  This will keep other checks running and let the user know that this one check configuration is invalid (and the check isn't running).

Check out [this code](https://github.com/Comcast/kuberhealthy/blob/2.0.0/cmd/kuberhealthy/kuberhealthy.go#L509) as an example here.  In this codepath, we build a `details` variable that contains the `khstate` of the check as we want it to be set.  Then, we call `k.storeCheckState()` on it to write it into the cluster's `khstate` resources.

I would expect a similar flow here when we fail to parse the configurations for the check.

It should be noted that if we do not continue on to run `k.AddCheck(c)` below, this check will _not_ appear on the status page or in metrics.  It will simply disappear because the `Kuberhealthy` struct isn't configured with it.  On the flip side, if we do run `k.AddCheck(c)`, then the check will end up being fed into the check runner code [here](https://github.com/Comcast/kuberhealthy/blob/2.0.0/cmd/kuberhealthy/kuberhealthy.go#L444).  Notice the first thing it does is setup a ticker with the `c.RunInterval`.

Perhaps we add an extra property to `Kuberhealthy` that contains a `map[string][]string`.  This would contain all the checks that should be skipped in each namespace (by name).  When a check fails to parse for any reason, we add its name to the list.  When running checks, we skip any check name found in the map.  Each time we re-configure Kuberhealthy's checks, we build a new map of checks to skip so not to get the map into a bad state.

This would make it possible to both add the check to the `Kuberhealthy.Checks` slice so it appears on the status page, AND to not attempt to actually execute the check.



So , to recap, here is how I propose this be written:
- Add a new `skippedChecks` `map[string][]string` property to `Kuberhealthy` and be sure it is initialized when checks are being added [here](https://github.com/Comcast/kuberhealthy/blob/2.0.0/cmd/kuberhealthy/kuberhealthy.go#L942) (`make map[string][]string`)
- Add a new method to `Kuberhealthy` that checks the new `skippedChecks` `map[string][]string` for a specific check in a specific namespace (just return a bool)
- Add a new method to `Kuberhealthy` that adds a check to be skipped to the `skippedChecks` namespace when given a namespace string and a check name string.
- Use the new method to decide if a check should be skipped on each run [here](https://github.com/Comcast/kuberhealthy/blob/2.0.0/cmd/kuberhealthy/kuberhealthy.go#L446).  Add a sleep and `continue` so it does not loop too fast.
- When a check [fails to parse something in around here](https://github.com/Comcast/kuberhealthy/blob/2.0.0/cmd/kuberhealthy/kuberhealthy.go#L338) during `configureChecks`, use the method you created to add the check by its namespace and name to the `skippedChecks` map.